### PR TITLE
feat: HTTP + HTTPS proxy support

### PR DESCRIPTION
**Pull request by BuethSam**
My workstation is behind a corporate proxy so I implemented proxy support. 

For that I had to switch to the request module. 

I also added a section about the possible config option which was written almost exclusively by ChatGPT. 😅

### CODE REVIEW

**Comment by BuethSam** [comment]
I used the https-proxy-agent module now to not have to reinvent the wheel. 'Import Cost' advertises it with 13.8k witch is indeed much better to the about 750k of 'request'. 
Yes still a dependency but much smaller.

**Comment by privatenumber** [comment]
That's fine. Thank you.

Can you also update it so it's passed in via config instead of the env var?

**Comment by BuethSam** [comment]
@privatenumber The proxy agent is now using a config option instead of the environment variable. I also added a section about the config options to the README

**Comment by privatenumber** [comment]
Would you mind pulling in the latest `develop`? Sorry for the conflicts!

I added tests so we can be more confident about changes.

File: README.md [hunk 1]
```diff
@@ -62,6 +62,24 @@ aicommits
...
+
+-   `proxy`: This option allows you to set a http/https proxy to use for requests. If you want to clear the proxy option, you can use the command aicommits config set proxy= (note the empty value after the equals sign).
+
+To set a configuration option, use the command `aicommits config set <key=value...>`. For example, to set the API key, you can use `aicommits config set OPENAI_KEY=<your-api-key>`. You can also set multiple configuration options at once by separating them with spaces, like `aicommits config set OPENAI_KEY=<your-api-key> generate=10 proxy=http://your-proxy-url`.
+
+To retrieve a configuration option, use the command `aicommits config get <key...>`. For example, to retrieve the API key, you can use `aicommits config get OPENAI_KEY`. You can also retrieve multiple configuration options at once by separating them with spaces, like `aicommits config get OPENAI_KEY generate`.
+
+Please note that setting any configuration option is optional and the CLI tool will still work without them.
+
+ > Fun fact: This section of the doc was written almost exclusively by ChatGPT.
```
Comment (privatenumber) [inline README.md:81]: 🔥🔥🔥

File: src/utils/config.ts [hunk 2]
```diff
@@ -51,6 +51,11 @@ const configParsers = {
 
 		return parsed;
 	},
+	proxy(key: string) {
+		parseAssert('proxy', key.length === 0 || /^https?:\/\//.test(key), 'Must be a valid URL');
```
Comment (privatenumber) [inline src/utils/config.ts:55]: I updated the logic here so if there's nothing provided, it returns the default value. So for this, you can probably just return if the argument is empty.

File: README.md (hunk 1 above)
Comment (privatenumber) [reply README.md:81]: Would you actually mind opening a separate PR for this? I can merge it pretty much immediately.

**Comment by privatenumber** [comment]
I guess we also need to support the `http_proxy` and `https_proxy` env vars:

https://github.com/Nutlope/aicommits/issues/136

File: README.md (hunk 1 above)
Comment (BuethSam) [reply README.md:81]: https://github.com/Nutlope/aicommits/pull/155 
Okay. I will wait for the merge of that PR and then rebase on that.

**Comment by privatenumber** [comment]
Will add tests in a followup PR

**Comment by BuethSam** [comment]
😍Thanks
using it daily now

**Pull request merged**